lua/nvim-tree/diagnostics.lua

```diff
@@ -58,7 +63,7 @@ function M.update()
     signs = {}
   end
   for bufname, severity in pairs(buffer_severity) do
-    if severity > 0 then
+    if 1 <= severity and severity <= 4 then
```

Comment:
i'd do if 0 < severity and severity < 5 then.
the 2nd check shouldn't happen though right ?

Reply:
Yes, the second expression is needed because `severity` might be `math.huge`. Possibly it could be a value higher than 4 as well, I don't know if severity 1-4 is just convention or if it is enforced by lsp. So we need to make sure `severity` is in the correct range otherwise `add_sign` will fail.